src/services/montage-helper.service.js

```diff
@@ -5,7 +5,27 @@
 		.module('montage')
 		.factory('montageHelper', montageHelper);
 
-	function montageHelper(montage, authService) {
+	function montageHelper(constants, montage, authService, ngProgressFactory) {
+
+		var pendingRequestCount = 0;
+		var progressBar = ngProgressFactory.createInstance();
+		progressBar.setColor(constants.primaryColor);
+
+		var _request = montage.Client.prototype.request;
+
+		montage.Client.prototype.request = function(...args) {
+			pendingRequestCount++;
+			progressBar.start();
+
+			return _request.bind(this)(...args).then(response => {
+				pendingRequestCount--;
+
+				if (!pendingRequestCount) { progressBar.complete(); }
```

Comment:
chaining a `.then()` on the existing one throws an error that `returnData` is undefined. Can we maybe look this over today?